src/redis_stack/general_data_structures.md

```diff
@@ -0,0 +1,320 @@
+
+```redis LPOP/RPOP usage
+LPOP bike:colors
+LRANGE bike:colors 0 -1
+RPOP bike:colors
+LRANGE bike:colors 0 -1
+LPOP bike:colors 2
+LRANGE bike:colors 0 -1
+KEYS bike:colors
+```
+
+**Note**:
+> `LRANGE` will return an empty (null) list if a key no longer exists.
+
+See [here](https://redis.io/commands/?group=list) for the entire set of Redis list-related commands.
+
+## Sets
+
+Redis [sets](https://redis.io/docs/data-types/sets/) are similar to lists, except they are unordered, and each element must be unique. You can use sets to:
+
+- Track unique items (e.g., track all unique IP addresses accessing a given blog post).
+- Represent relations (e.g., the set of all users with a specified role).
+- Perform common set operations such as intersection, union, and difference.
+
+Here are some of the important set commands:
+
+- [SADD](https://redis.io/commands/sadd)
+- [SREM](https://redis.io/commands/srem)
+- [SISMEMBER](https://redis.io/commands/sismember)
+- [SMEMBERS](https://redis.io/commands/smembers)
+- [SUNION](https://redis.io/commands/sunion)
+
+Use `SADD` to create and update a set. Each `SADD` command will return the number of added members. If you try to add a member that is already present in the set, `0` is returned.
+
+```redis Create a set
+SADD bike:1:addons "whitewall tires"
+SADD bike:1:addons "bell" "reflectors"
+SADD bike:1:addons "bell"
+```
+
+Notice that the `SADD` command is variadic.
+
+`SREM` is used to remove members of a list. It returns `1` if the member is present in the set, or `0` if it is not.
+
+```redis Remove set members
+SREM bike:1:addons "bell"
+SREM bike:1:addons "sissy bar"
+```
+
+The `SISMEMBER` command is used to test set membership. It returns `1` if the member is present, and `0` otherwise.
+
+```redis Test set membership
+SISMEMBER bike:1:addons "reflectors"
+SISMEMBER bike:1:addons "sissy bar"
+```
+
+Use `SMEMBERS` to return all the members of a set.
+
+```redis Get set members
+SMEMBERS bike:1:addons
+```
+
+`SUNION` combines two or more sets and returns all their elements.
+
+```redis SUNION usage
+SADD bike:1:special_addons "sparkle coat finish" "banana seat"
+SUNION bike:1:addons bike:1:special_addons
+```
+
+Sets have two ways to return one or more random members of a set:
+
+- [SPOP](https://redis.io/commands/spop)
+- [SRANDMEMBER](https://redis.io/commands/srandmember)
+
+Each command takes a key and, optionally, a count as arguments. `SPOP` removes and returns a random member, whereas `SRANDMEMBER` just returns the randomly selected members without removing them.
+
+```redis SPOP/SRANDMEMBER usage
+SMEMBERS bike:1:addons
+SRANDMEMBER bike:1:addons
+SPOP bike:1:addons
+SMEMBERS bike:1:addons
+```
+
+See [here](https://redis.io/commands/?group=set) for the entire list of Redis set-related commands.
+
+## Sorted sets
+
+While handy, Redis sets are unsorted, which limits their usefulness. It is for this reason that Redis sorted sets were added to its stable of data types.
```

Comment:
I think the second sentence would benefit from rewording. "It is for this reason that" seems a little wordy, and I'm not sure what "its stable of data types" means. Is there a word missing here?